shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -434,7 +437,90 @@ private int getPointerDeviceTypeForToolType(int toolType) {
         }
     }
 
-    private void addPointerForIndex(MotionEvent event, int pointerIndex, int pointerChange,
+    private List<PointerAction> expandActions(MotionEvent event, int pointerIndex, int pointerChange,
+    int batched) {
+        long device = event.getPointerId(pointerIndex);
+        // the android touch events do not always follow clear
+        // guideline. We normalize the events so that the framework
+        // does not have to deal with android specific issue.
+        double dx = event.getX(pointerIndex);
+        double dy = event.getY(pointerIndex);
+        PointerState state;
+        List<PointerAction> requiredAction = new ArrayList<PointerAction>();
+        switch (pointerChange) {
+            case kPointerChangeCancel:
+            case kPointerChangeUp:
+                // Some android phone will send a cancel request to non exist point
```

Comment:
phone -> phones
request -> event

at the end: `...for a non-existent pointer,`